MekHQ/src/mekhq/campaign/universe/RATGeneratorConnector.java

```diff
@@ -53,126 +52,124 @@ public RATGeneratorConnector(int year) {
+                                      @Nullable Predicate<MechSummary> filter) {
+        final UnitTable table = findTable(faction, unitType, weightClass, year, quality, movementModes, missionRoles);
+        return (table == null) ? new ArrayList<>() : table.generateUnits(count, (filter == null) ? null : filter::test);
     }
 
     /**
      * Generates a list of mech summaries from a RAT determined by the given faction, quality and other parameters.
-     * Note that for the purposes of this implementation, any faction record or unit quality are ignored.
+     * We force a fallback to try to ensure that something is generated if the parents have any possible units to generate,
+     * as that is the normally expected behaviour for MekHQ OpFor generation.
      * @param count How many units to generate
      * @param parameters RATGenerator parameters
      */
     @Override
-    public List<MechSummary> generate(int count, UnitGeneratorParameters parameters) {
-        UnitTable ut = UnitTable.findTable(parameters.getRATGeneratorParameters());
-
-        return ut == null ? new ArrayList<MechSummary>() :
-            ut.generateUnits(count, parameters.getFilter() == null ? null : ms -> parameters.getFilter().test(ms));
+    public List<MechSummary> generate(final int count, final UnitGeneratorParameters parameters) {
+        final UnitTable table = findOpForTable(parameters);
+        return table.generateUnits(count, (parameters.getFilter() == null) ? null : ms -> parameters.getFilter().test(ms));
     }
 
     /**
      * Generates a single mech summary from a RAT determined by the given faction, quality and other parameters.
-     * Note that for the purposes of this implementation, any faction record or unit quality are ignored.
+     * We force a fallback to try to ensure that something is generated if the parents have any possible units to generate,
+     * as that is the normally expected behaviour for MekHQ OpFor generation.
      * @param parameters RATGenerator parameters
      */
     @Override
-    public MechSummary generate(UnitGeneratorParameters parameters) {
-        UnitTable ut = UnitTable.findTable(parameters.getRATGeneratorParameters());
+    public @Nullable MechSummary generate(final UnitGeneratorParameters parameters) {
+        final UnitTable table = findOpForTable(parameters);
+        return table.generateUnit((parameters.getFilter() == null) ? null : ms -> parameters.getFilter().test(ms));
+    }
 
-        return ut == null ? null :
-            ut.generateUnit(parameters.getFilter() == null ? null : ms -> parameters.getFilter().test(ms));
+    /**
+     * This finds a unit table for OpFor generation. It falls back using the parent faction to try to ensure there are
+     * units in the unit table, so an OpFor is generated.
+     * @param unitParameters the base parameters to find the table using.
+     * @return the unit table to generate using
```

Comment:
A bit unclear as to what `using` relates to.

Reply:
Will clarify in the morrow